cf-custom-resources/lib/custom-domain-app-runner.js

```diff
@@ -0,0 +1,304 @@
+let appRoute53Client, appRunnerClient, appHostedZoneID;
+
+/**
+ * Upload a CloudFormation response object to S3.
+ *
+ * @param {object} event the Lambda event payload received by the handler function
+ * @param {object} context the Lambda context received by the handler function
+ * @param {string} responseStatus the response status, either 'SUCCESS' or 'FAILED'
+ * @param {string} physicalResourceId CloudFormation physical resource ID
+ * @param {object} [responseData] arbitrary response data object
+ * @param {string} [reason] reason for failure, if any, to convey to the user
+ * @returns {Promise} Promise that is resolved on success, or rejected on connection error or HTTP error response
+ */
+function report (
+    event,
+    context,
+    responseStatus,
+    physicalResourceId,
+    responseData,
+    reason
+) {
+    return new Promise((resolve, reject) => {
+        const https = require("https");
+        const { URL } = require("url");
+
+        let reasonWithLogInfo = `${reason} (Log: ${context.logGroupName}/${context.logStreamName})`;
+        var responseBody = JSON.stringify({
+            Status: responseStatus,
+            Reason: reasonWithLogInfo,
+            PhysicalResourceId: physicalResourceId || context.logStreamName,
+            StackId: event.StackId,
+            RequestId: event.RequestId,
+            LogicalResourceId: event.LogicalResourceId,
+            Data: responseData,
+        });
+
+        const parsedUrl = new URL(event.ResponseURL);
+        const options = {
+            hostname: parsedUrl.hostname,
+            port: 443,
+            path: parsedUrl.pathname + parsedUrl.search,
+            method: "PUT",
+            headers: {
+                "Content-Type": "",
+                "Content-Length": responseBody.length,
+            },
+        };
+
+        https
+            .request(options)
+            .on("error", reject)
+            .on("response", (res) => {
+                res.resume();
+                if (res.statusCode >= 400) {
+                    reject(new Error(`Error ${res.statusCode}: ${res.statusMessage}`));
+                } else {
+                    resolve();
+                }
+            })
+            .end(responseBody, "utf8");
+    });
+}
+
+exports.handler = async function (event, context) {
+    const props = event.ResourceProperties;
+    const [serviceARN, appDNSRole, customDomain] = [props.ServiceARN, props.AppDNSRole, props.CustomDomain,];
+    appHostedZoneID = props.HostedZoneID;
+    let physicalResourceID = event.PhysicalResourceId || event.LogicalResourceId;
+
+    // Report with failure if lambda times out.
+    let lambdaTimedOut = lambdaTimeOut.then(async () => {
+        console.log(`Lambda is about to time out. Operation fails.`);
+        // throw new Error("Lambda about to time out");
+        await report(event, context, "FAILED", physicalResourceID, null, "Lambda about to time out");
+    });
+
+    let handler = async function () {
```

Comment:
What do you think of refactoring such that we remove all the `report` calls from `handler`? then the main function can be read as:
```js
let handler = async () => { ... }

try {
  await Promise.race(handler, deadlineExpired());
  await report(event, context, "SUCCESS", physicalResourceID);
} catch (err) {
  if (err.name === ERR_NAME_INVALID_REQUEST && ...) {
    await report(event, context, "SUCCESS", physicalResourceID);
    return;
  } 
  await report(event, context, "FAILED", physicalResourceID, null, err.message);
}


```

Reply:
yesss it was refactored this way! i pushed out the changes too soon.